### PR TITLE
Conform more of the codebase to strictNullChecks

### PR DESCRIPTION
**Pull request by t3chguy**
For https://github.com/vector-im/element-web/issues/21967

---
This change is marked as an *internal change* (Task), so will not be included in the changelog.

### CODE REVIEW

File: src/utils/SortMembers.ts [hunk 1]
```diff
@@ -96,13 +97,14 @@ interface IMemberScore {
     numRooms: number;
 }
 
-export function buildMemberScores(cli: MatrixClient): { [key: string]: IMemberScore | undefined } {
+export function buildMemberScores(cli: MatrixClient): { [key: string]: IMemberScore } {
```
Comment (alunturner) [inline src/utils/SortMembers.ts:100]: Minor point - not sure if we prefer `<Record<string, ...>` for cases like this (also valid elsewhere in this file)

**Review by alunturner — COMMENTED** [review]
Changes look good to me. One small comment and also a question about Searching.ts - are we expecting this to be completely clear of TS issues? I see a few TS errors there.

File: src/utils/SortMembers.ts (hunk 1 above)
Comment (t3chguy) [reply src/utils/SortMembers.ts:100]: In a case where the key string is uselessly `key` I agree, but IMO this form is better as the key can be given a helpful name, I'll update it to `userId` for self-documenting

**Comment by t3chguy** [comment]
> are we expecting this to be completely clear of TS issues? I see a few TS errors there.

Not yet, not the best way to do things but still helps the overall goal

File: src/utils/SortMembers.ts (hunk 1 above)
Comment (alunturner) [reply src/utils/SortMembers.ts:100]: I'd never considered that, makes sense

**Review by alunturner — APPROVED** [review]

**Pull request merged**